BenchPress/Helpers/BenchPress.Azure/Common.psm1

```diff
@@ -228,12 +254,20 @@ function Confirm-Resource {
     $ConfirmResult = [ConfirmResult]::new($ErrorRecord, $null)
   } elseif ($ConfirmResult.Success) {
     if ($PropertyKey) {
-      if ($ConfirmResult.ResourceDetails.$PropertyKey -ne $PropertyValue) {
-        $ConfirmResult.Success = $false
-        $ConfirmResult.ErrorRecord = `
-          Format-IncorrectValueError -ExpectedKey $PropertyKey `
-                                     -ExpectedValue $PropertyValue `
-                                     -ActualResult $ConfirmResult.ResourceDetails.$PropertyKey
+      $ActualValue = $ConfirmResult.ResourceDetails
+      $Keys = $PropertyKey.Split(".")
+      foreach($Key in $Keys){
+        $ActualValue = $ActualValue.$Key
+      }
+      if ($ActualValue -ne $PropertyValue) {
```

Comment:
Is this if-statement supposed to be in the the foreach block to check each of the values?

Reply:
No. The foreach navigates the key path to get to the final property. Only the value of the last property is evaluated.

For example, if the initial "object" is `{"x":{"y":{"z":"finalValue"}}}` and the path were `x.y.z`, then the `$ActualValue` would be the value of property `z` which would be `"finalValue"`. That's the only value that matters for the `if` block.